ddclient.in

```diff
@@ -2155,6 +2168,171 @@ sub geturl {
     return $reply;
 }
 
+######################################################################
+## curl_cmd() function to execute system curl command
+######################################################################
+sub curl_cmd {
+    my @params = @_;
+    my $tmpfile;
+    my $tfh;
+    my $system_curl = subst_var('@syscurlcmd@', '/usr/bin/curl');
+    my %curl_codes = (  ## Subset of error codes from https://curl.haxx.se/docs/manpage.html
+         2 => "Failed to initialize. (Most likely a bug in ddclient, please open issue at https://github.com/ddclient/ddclient)",
+         3 => "URL malformed. The syntax was not correct",
+         5 => "Couldn't resolve proxy. The given proxy host could not be resolved.",
+         6 => "Couldn't resolve host. The given remote host was not resolved.",
+         7 => "Failed to connect to host.",
+        22 => "HTTP page not retrieved. The requested url was not found or returned another error.",
+        28 => "Operation timeout. The specified time-out period was reached according to the conditions.",
+        35 => "SSL connect error. The SSL handshaking failed.",
+        47 => "Too many redirects. When following redirects, curl hit the maximum amount.",
+        51 => "The peer's SSL certificate or SSH MD5 fingerprint was not OK.",
+        58 => "Problem with the local certificate.",
+        60 => "Peer certificate cannot be authenticated with known CA certificates.",
+        67 => "The user name, password, or similar was not accepted and curl failed to log in.",
+        77 => "Problem with reading the SSL CA cert (path? access rights?).",
+        78 => "The resource referenced in the URL does not exist.",
+       127 => "You requested network access with curl but $system_curl was not found",
+    );
+
+    return '' if (scalar(@params) == 0); ## no parameters provided
+
+    # debug("CURL: %s\r\n",join("\r\n      ",@params)); ## Uncomment only if required as may expose credentials
+
+    ## Use system File::Temp to generate filename...
+    use File::Temp;
+    $tfh = File::Temp->new( TMPDIR => 1,  ## in system temp dir
+                            UNLINK => 0,  ## we will delete it ourselves
```

Comment:
I was getting warning messages that the file had already been deleted on exiting the function that I could not suppress.  This was potentially relayed to handling SIGINT etc. and because I had alternatives to File::Temp.  But as we're going to remove all that then we can probably revert back to auto deletion -- I will test and update change it if there are no problems.

Reply:
I have updated to use the built-in feature of File::Temp to automatically delete it.